xamarin/realm-tutorial-dotnet/LoginPage.xaml.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using AsyncTask = System.Threading.Tasks.Task;
+using Realms.Sync;
+using Xamarin.Forms;
+
+namespace RealmTemplateApp
+{
+    public partial class LoginPage : ContentPage
+    {
+        private string email;
+        private string password;
+
+        public LoginPage()
+        {
+            InitializeComponent();
+        }
+
+        protected override void OnAppearing()
+        {
+            base.OnAppearing();
+            txtEmail.Text = "";
+            txtPassword.Text = "";
+        }
+
+
+        async void Login_Button_Clicked(object sender, EventArgs e)
+        {
+            await DoLogin();
+        }
+
+        private async AsyncTask DoLogin()
+        {
+            try
+            {
+                var user = await App.RealmApp.LogInAsync(Credentials.EmailPassword(email, password));
+                if (user != null)
+                {
+                    var taskPage = new TaskPage();
+                    NavigationPage.SetHasBackButton(taskPage, false);
+                    await Navigation.PushAsync(taskPage);
+                }
+                else throw new Exception();
```

Comment:
This flow feels a bit odd. Wouldn't it actually be an assertion (i.e. dev error, not user-facing runtime exception) that user is not null at this point?

Reply:
Again I'm not sure what you are referring to. There's nothing in the code I see here that has to do with a null user.

Reply:
Line 35 gets the user by logging in. Presumably this method throws on error (no network, bad password, etc). But we then also check that the user is null. In what case does LogInAsync return null?